### ISSUE TITLE
[Train] Better debuggability when training is hanging

### ISSUE THREAD

**Issue by amogkam** [ray-project/ray#22567]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/ray/issues) and found no similar feature requirement.

### Description

When Ray Train hangs, it's very hard to get an understanding on what it's hanging on because the main process is just waiting on a `ray.get` call. Debugging this requires multiple `py-spy` calls (possibly on multiple nodes in the cluster) and then interpreting the output.

We can make the first part easier by adding a utility to call `py-spy` on the all nodes in the cluster for the processes that are relevant to Ray Train (main process, BackendExecutor, BaseWorkerMixin).

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by scottsun94** [comment]
Is this similar to `ray stacks` but specific for Ray Train? Could we actually extend `ray stacks` to support it?
cc: @rkooo567

**Comment by amogkam** [comment]
That's right it's basically calling `ray stack`, but also making it more interpretable for a Ray Train user.

**Comment by rkooo567** [comment]
We are thikning to support running py-spy on running workers in 2.2. Would that satisfy the req?

**Comment by rkooo567** [comment]
I already have a prototype working

**Comment by scottsun94** [comment]
In 2.2, @rkooo567 's prototype has been added to the dashboard. 
![Screen Shot 2022-12-28 at 2 42 33 PM](https://user-images.githubusercontent.com/9677264/209881707-7e2904b0-fe1b-4563-9de1-6e44bee15a50.png)

Do you think it meets your requirements? @amogkam

**Comment by rkooo567** [comment]
Stack trace should cover this. If there's a usability issue, please create another issue. I will close the issue for now
